indexing-service/src/main/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTask.java

```diff
@@ -203,11 +208,9 @@ public TaskStatus runTask(TaskToolbox toolbox) throws Exception
     );
     // Fetch the load specs of all segments overlapping with the unused segment intervals
```

Comment:
With this patch, the kill task logic for selecting segments to address correctness has become fairly complex. Could we add some high-level commentary on the overall approach for the `runTask()` method? The line comments added provide specifics, which is nice, but I think a method javadoc would provide a better mental model for a reader before digging into the details.

Reply:
+1, the class level javadoc would be even better.

Reply:
Done